src/dynatrace/opentelemetry/metrics/export/__init__.py

```diff
@@ -37,6 +39,87 @@
 VERSION = "0.2.0b0"
 
 
+def _get_histogram_max(histogram: Histogram):
+    if math.isfinite(histogram.max):
+        return histogram.max
+
+    histogram_sum = histogram.sum
+    histogram_count = sum(histogram.bucket_counts)
+    if len(histogram.bucket_counts) == 1:
+        # In this case, only one bucket exists: (-Inf, Inf). If there were
+        # any boundaries, there would be more counts.
+        if histogram.bucket_counts[0] > 0:
+            # in case the single bucket contains something, use the mean as
+            # max.
+            return histogram_sum / histogram_count
+        # otherwise the histogram has no data. Use the sum as the min and
+        # max, respectively.
+        return histogram_sum
+
+    # loop over bucket_counts in reverse
+    last_element_index = len(histogram.bucket_counts) - 1
+    for index in range(last_element_index, -1, -1):
+        if histogram.bucket_counts[index] > 0:
+            if index == last_element_index:
+                # use the last bound in the bounds array. This can only be
+                # the case if there is a count > 0 in the last bucket (
+                # lastBound, Inf), therefore, the bound has to be smaller
+                # than the actual maximum value, which in turn ensures that
+                # the sum is larger than the bound we use as max here.
+                return histogram.explicit_bounds[index - 1]
+            # in any bucket except the last, make sure the sum is greater
+            # than or equal to the max, otherwise report the sum.
+            return min(histogram.explicit_bounds[index], histogram_sum)
+
+    # there are no counts > 0, so calculating a mean would result in a
+    # division by 0. By returning the sum, we can let the backend decide what
+    # to do with the value (with a count of 0)
+    return histogram_sum
+
+
+def _get_histogram_min(histogram: Histogram):
+    if math.isfinite(histogram.min):
+        return histogram.min
+
+    histogram_sum = histogram.sum
+    histogram_count = sum(histogram.bucket_counts)
+    if len(histogram.bucket_counts) == 1:
```

Comment:
@joaopgrassi in .NET do you check that bucket counts length is 0 or bucket boundaries is length 0? It should not be possible to have counts == 0 because counts is always boundaries + 1

Reply:
Hum... I think it's the number of bucket boundaries. 

The `if (buckets.Count == 0)` variable it uses, is obtain by getting the histogram buckets here https://github.com/dynatrace-oss/opentelemetry-metric-dotnet/blob/main/src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs#L71

It throws me off a bit, but I think it's correct. Marc probably knows best as I think he also implemented/improved the .NET impl of the estimation

Reply:
In .NET they have a different layout. A `MetricPoint` has an array of `HistogramBucket`s where `ExplicitBoundary` and `BucketCount` are grouped together. With a catch-all bucket (`-Inf` to `+Inf`) will have an empty array of buckets. The `MetricPoint` itself has methods `GetHistogramSum()` and `GetHistogramCount()`, which is then used instead for calculating the average, when that array is empty. This is why we check for `buckets.Count == 0` there instead of `==1`.

In Python, we have `bucket_counts` and `explicit_bounds`. There `bucket_counts` always has at least `1` element, but `explicit_bounds` would have `0` elements. Here, in fact checking `len(histogram.bucket_counts) == 1` indicates that there is only one bucket. This is the same approach as in the Java implementation of this algorithm. :slightly_smiling_face: